include/Gaffer/ValuePlug.h

```diff
@@ -205,6 +205,14 @@ class GAFFER_API ValuePlug : public Plug
 
 		//@}
 
+		/// Returns the current dirty count of this plug.
+		/// This is usually only used internally, but is occasionally useful in nodes which have specialized
+		/// ways of storing things that may be invalidated, like Encapsulate
+		uint64_t dirtyCount() const
```

Comment:
Would be great to add a Python binding for this, as I think it might be quite handy for interactive debugging.

Reply:
I did this and added tests, however these tests currently do not pass - because I would expect parenting a plug to update the dirtyCount?  This is probably just a case of you telling me why my expectation is wrong, and me updating the test.  But it currently seems quite odd to me that reparenting emits plugSet but doesn't update the dirtyCount, and I probably shouldn't update the test to the current behaviour until I understand why.

Reply:
I think you need to parent the plug to a node rather than another plug - we need a node for dirty propagation to be meaningful.